### DISCUSSION TITLE
How to configure 2.23.0+ AbstractLogger to not use ParameterizedMessageFactory?

### DISCUSSION THREAD

**Discussion by JohntheMorty** [apache/logging-log4j2#2838] [Q&A]
Hi, I noticed since 2.23.0 there was a change #2249 which looks like `AbstractLogger` no longer selects `MessageFactory` between `ReusableParameterizedMessageFactory` and `ParameterizedMessageFactoryClass` based on `Constants.ENABLE_THREADLOCALS`.

Does this means `log4j2.enable.threadlocals` is no longer used to select `ReusableParameterizedMessageFactory` for GC free logging? What's the recommanded way now to select `ReusableParameterizedMessageFactory` for the logger so that my class that implements `StringBuilderFormattable` can have its `formatTo(StringBuilder buffer)` method called instead of `toString()`?

Thanks heaps!

**Comment by ppkarwasz** [comment]
Hi @JohntheMorty,

> Does this means `log4j2.enable.threadlocals` is no longer used to select `ReusableParameterizedMessageFactory` for GC free logging? 

No, it is simply a regression (#2379) that will be fixed in the upcoming 2.24.0 version.

Log4j API 2.22.x and earlier version had a complex initialization process that required the usage of **two** classes (`StatusLogger` and `LowLevelLogUtil`) to convey internal logging messages. The initialization process has been simplified in 2.23.x, but it is now up to the implementors of `AbstractLogger` to pass the correct message factories as constructor parameters.
See [API Initialization](https://cwiki.apache.org/confluence/display/LOGGING/API+Initialization) for details.
